ndn_hydra/client/functions/query.py

```diff
@@ -32,12 +32,14 @@ async def send_query(self, query: Name, node_name: str=None) -> None:
       Form a certain query and request that info from a node.
       """
       if not node_name:
-          named_query = self.repo_prefix + [Component.from_str("query")] + query
+        named_query = self.repo_prefix + [Component.from_str("query")]
       else:
-          named_query = self.repo_prefix + [Component.from_str("node")] + [Component.from_str(node_name)] + [Component.from_str("query")] + query
-
+        named_query = self.repo_prefix + [Component.from_str("node")] + [Component.from_str(node_name)] + [Component.from_str("query")]
+      querytype = query[0]

```

Comment:
I am unsure why you need to specify the querytype here.

Just do app_param = query[0].
Querytype is redefined later on and used afterwards which works.

Reply:
That makes sense. Resolved in most recent commit.